tsdb/head_test.go

```diff
@@ -402,6 +402,39 @@ func TestHead_WALMultiRef(t *testing.T) {
 	}}, series)
 }
 
+func TestHead_ActiveAppenders(t *testing.T) {
+	head, _ := newTestHead(t, 1000, false)
+	defer head.Close()
+
+	require.NoError(t, head.Init(0))
+
+	// First rollback with no samples
+	app := head.Appender(context.Background())
+	require.Equal(t, 1.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+	require.NoError(t, app.Rollback())
+	require.Equal(t, 0.0, prom_testutil.ToFloat64(head.metrics.activeAppenders))
+
+	// Then commit with no samples
+	app = head.Appender(context.Background())
+	require.NoError(t, app.Commit())
```

Comment:
Like for rollback, we should check if it was 1.0 just before this line

Reply:
Under what model could the behaviour vary between the two cases?